workflow/artifacts/s3/s3.go

```diff
@@ -29,7 +31,23 @@ type ArtifactDriver struct {
 	Context     context.Context
 }
 
-var _ artifactscommon.ArtifactDriver = &ArtifactDriver{}
+// s3TransientErrorCodes is a list of S3 error codes that are transient (retryable)
+// Reference: https://github.com/minio/minio-go/blob/92fe50d14294782d96402deb861d442992038109/retry.go#L90-L102
+var s3TransientErrorCodes = []string{
```

Comment:
minor - could you move to own file?

Reply:
Sure. Moved this to `error_codes.go` in f5c55fcdcd0a0390269d9328c63ad56090c5d3cc. Thanks!